src/home/sorting/sorting-manager.ts

```diff
@@ -50,22 +54,36 @@ export class SortingManager {
     textBox.value = searchQuery;
 
     const issuesContainer = document.getElementById("issues-container") as HTMLDivElement;
-
-    function filterIssues() {
+    const filterIssues = () => {
       try {
         const filterText = textBox.value.toLowerCase();
         const issues = Array.from(issuesContainer.children) as HTMLDivElement[];
-        issues.forEach((issue) => {
-          const issueId = issue.children[0].getAttribute("data-issue-id");
-          issue.classList.add("active");
-          if (!issueId) return;
-          const gitHubIssue = taskManager.getGitHubIssueById(parseInt(issueId));
-          if (!gitHubIssue) return;
-          const searchableProperties = ["title", "body", "number", "html_url"] as const;
-          const searchableStrings = searchableProperties.map((prop) => gitHubIssue[prop]?.toString().toLowerCase());
-          const isVisible = searchableStrings.some((str) => str?.includes(filterText));
-          issue.style.display = isVisible ? "block" : "none";
-        });
+        // Get issue IDs and search results
+        const issueIds = issues
+          .map((issue) => issue.children[0].getAttribute("data-issue-id"))
+          .filter((id): id is string => id !== null)
+          .map((id) => parseInt(id));
+
+        const searchResults = taskManager.issueSearcher.search(filterText, issueIds);
+        // If there's a search term, sort by relevance
+        if (filterText) {
+          issues
+            .sort((a, b) => {
```

Comment:
would you mind writing a function in another file to abstract this code section? this is how we do it for other sorting criteria:

![image](https://github.com/user-attachments/assets/fccd410e-22be-42d7-a5b7-e0dc3cb4b2a2)